src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.DateTime.netfx.cs

```diff
@@ -42,6 +42,32 @@ private class TimeSync
         private static TimeSync timeSync = new TimeSync();
 
         // sync DateTime and Stopwatch ticks every 2 hours
-        private static Timer syncTimeUpdater = new Timer(s => { Sync(); }, null, 0, 7200000);
+        private static Timer syncTimeUpdater = InitalizeSyncTimer();
+
+        [System.Security.SecuritySafeCritical]
+        private static Timer InitalizeSyncTimer()
+        {
+            Timer timer;
+            // Don't capture the current ExecutionContext and its AsyncLocals onto the timer causing them to live forever
+            bool restoreFlow = false;
+            try
+            {
+                if (!ExecutionContext.IsFlowSuppressed())
+                {
+                    ExecutionContext.SuppressFlow();
+                    restoreFlow = true;
+                }
+
+                timer = new Timer(s => { Sync(); }, null, 0, 7200000);
```

Comment:
Nit: how about just returning it here rather than defining the local above and returning it below... one line instead of three.